app/javascript/Components/__tests__/table.test.jsx

```diff
@@ -0,0 +1,359 @@
+      expect(document.querySelector(".Table.-highlight")).toBeInTheDocument();
+      expect(document.querySelector(".rt-table")).toBeInTheDocument();
+      expect(document.querySelector(".rt-thead.-header")).toBeInTheDocument();
+      expect(document.querySelector(".rt-thead.-filters")).toBeInTheDocument();
+      expect(document.querySelector(".rt-tbody")).toBeInTheDocument();
+    });
+
+    it("renders rows in correct order", () => {
+      const {container} = render(<Table columns={mockColumns} data={mockData} />);
+
+      const rows = container.querySelectorAll(".rt-tbody .rt-tr");
+      expect(rows).toHaveLength(4);
+
+      expect(rows[0]).toHaveTextContent("John Doe");
+      expect(rows[1]).toHaveTextContent("Jane Smith");
+      expect(rows[2]).toHaveTextContent("Bob Johnson");
+      expect(rows[3]).toHaveTextContent("Alice Brown");
+    });
+  });
+
+  describe("No Data Handling", () => {
+    it("displays default no data message when data is empty", () => {
+      render(<Table columns={mockColumns} data={[]} />);
+
+      expect(screen.getByText("No rows found")).toBeInTheDocument();
+      expect(screen.getByText("No rows found")).toHaveClass("rt-no-data");
+    });
+
+    it("displays custom no data message", () => {
+      const customMessage = "No results available";
+      render(<Table columns={mockColumns} data={[]} noDataText={customMessage} />);
+
+      expect(screen.getByText(customMessage)).toBeInTheDocument();
+      expect(screen.queryByText("No rows found")).not.toBeInTheDocument();
+    });
+
+    it("does not display no data message when data exists", () => {
+      render(<Table columns={mockColumns} data={mockData} />);
+
+      expect(screen.queryByText("No rows found")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("Column Sorting", () => {
+    it("sorts column when header is clicked", async () => {
+      const user = userEvent.setup();
+      render(<Table columns={mockColumns} data={mockData} />);
+
+      const nameHeader = screen.getByText("Name");
+      await user.click(nameHeader);
+
+      const headerElement = nameHeader.closest(".rt-th");
+      expect(headerElement).toHaveClass("--cursor-pointer");
+    });
+
+    it("applies correct sorting classes", async () => {
+      const user = userEvent.setup();
+      render(<Table columns={mockColumns} data={mockData} />);
+
+      const nameHeader = screen.getByText("Name");
+
+      await user.click(nameHeader);
+      expect(nameHeader.closest(".rt-th")).toHaveClass("rt-resizable-header-content");
+
+      await user.click(nameHeader);
+      expect(nameHeader.closest(".rt-th")).toHaveClass("rt-resizable-header-content");
+    });
+
+    it("applies sorting classes to all headers", () => {
+      render(<Table columns={mockColumns} data={mockData} />);
+
+      const emailHeader = screen.getByText("Email");
+      const headerElement = emailHeader.closest(".rt-th");
+
+      expect(headerElement).toHaveClass("--cursor-pointerundefined");
```

Comment:
the `undefined` is quite janky; please modify the implementation so that no class is added if the sort order isn't `asc`/`desc`